test-framework/maven/src/main/java/io/quarkus/maven/it/verifier/RunningInvoker.java

```diff
@@ -20,27 +30,58 @@
     private final boolean debug;
     private MavenProcessInvocationResult result;
     private final File log;
-    private final PrintStreamHandler logHandler;
+    private final PrintStreamHandler outStreamHandler;
+    private final PrintStreamHandler errStreamHandler;
 
+    /**
+     * This is the same as calling {@link #RunningInvoker(File, boolean, boolean)} with
+     * {@code redirectToLogFile} as {@code false}
+     */
     public RunningInvoker(File basedir, boolean debug) throws FileNotFoundException {
+        this(basedir, debug, false);
+    }
+
+    /**
+     * Creates a {@link RunningInvoker}
+     *
+     * @param basedir The base directory to use for invoking the Maven process
+     * @param debug Whether or not the Maven process that's launched should use the "debug" flag
+     * @param redirectToLogFile If {@code true} then the launched Maven process will send its
+     *        logs and errors to a log file. Else, the logs are streamed to the
+     *        {@code System.out} and {@code System.err} of this JVM's process
+     * @throws FileNotFoundException
+     */
+    public RunningInvoker(final File basedir, final boolean debug, final boolean redirectToLogFile)
+            throws FileNotFoundException {
         this.debug = debug;
         setWorkingDirectory(basedir);
         String repo = System.getProperty("maven.repo");
         if (repo == null) {
             repo = new File(System.getProperty("user.home"), ".m2/repository").getAbsolutePath();
         }
         setLocalRepositoryDirectory(new File(repo));
-        log = new File(basedir, "build-" + basedir.getName() + ".log");
-        PrintStream stream = null;
-        try {
-            stream = new PrintStream(log, "UTF-8");
-        } catch (UnsupportedEncodingException e) {
-            stream = new PrintStream(log);
+        final PrintStream outStream;
+        final PrintStream errStream;
+        if (redirectToLogFile) {
+            log = new File(basedir, "build-" + basedir.getName() + ".log");
+            PrintStream stream = null;
+            try {
+                stream = new PrintStream(log, "UTF-8");
+            } catch (UnsupportedEncodingException e) {
+                stream = new PrintStream(log);
+            }
+            outStream = errStream = stream;
+        } else {
+            log = null;
+            outStream = System.out;
+            errStream = System.err;
         }
-        logHandler = new PrintStreamHandler(stream, true);
-        setErrorHandler(logHandler);
-        setOutputHandler(logHandler);
-        setLogger(new PrintStreamLogger(stream, InvokerLogger.DEBUG));
+        outStreamHandler = new PrintStreamHandler(outStream, true);
+        setOutputHandler(outStreamHandler);
+
+        errStreamHandler = new PrintStreamHandler(errStream, true);
+        setErrorHandler(errStreamHandler);
+        setLogger(new PrintStreamLogger(outStream, InvokerLogger.DEBUG));
```

Comment:
Should we be picking the level based on the value of `debug` argument here?

Reply:
Hello @aloubyansky ,

That sounds reasonable. I've updated the PR to include your suggestion.